core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala

```diff
@@ -22,297 +22,257 @@ import java.util.concurrent.TimeUnit
         assignments.get.filter(_.group == group).head.host.exists(_.trim != ConsumerGroupCommand.MISSING_COLUMN_VALUE)
     }, "Expected a 'Stable' group status, rows and valid values for consumer id / client id / host columns in describe group results.")
-
-    consumerGroupCommand.close()
-    executor.shutdown()
   }
 
   @Test
   def testDescribeExistingGroupWithNoMembersWithNewConsumer() {
+    TestUtils.createOffsetsTopic(zkUtils, servers)
     // run one consumer in the group consuming from a single-partition topic
-    val executor = new ConsumerGroupExecutor(brokerList, 1, group, topic)
+    consumerGroupExecutor = new ConsumerGroupExecutor(brokerList, 1, group, topic)
 
     val cgcArgs = Array("--bootstrap-server", brokerList, "--describe", "--group", group)
     val opts = new ConsumerGroupCommandOptions(cgcArgs)
-    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+    consumerGroupService = new KafkaConsumerGroupService(opts)
 
     TestUtils.waitUntilTrue(() => {
-      val (state, _) = consumerGroupCommand.describeGroup()
+      val (state, _) = consumerGroupService.describeGroup()
       state == Some("Stable")
     }, "Expected the group to initially become stable.")
 
     // stop the consumer so the group has no active member anymore
-    executor.shutdown()
-
-    TestUtils.waitUntilTrue(() => {
-        val (state, assignments) = consumerGroupCommand.describeGroup()
-        state == Some("Empty") &&
-        assignments.isDefined &&
-        assignments.get.count(_.group == group) == 1 &&
-        assignments.get.filter(_.group == group).head.consumerId.exists(_.trim == ConsumerGroupCommand.MISSING_COLUMN_VALUE) && // the member should be gone
-        assignments.get.filter(_.group == group).head.clientId.exists(_.trim == ConsumerGroupCommand.MISSING_COLUMN_VALUE) &&
-        assignments.get.filter(_.group == group).head.host.exists(_.trim == ConsumerGroupCommand.MISSING_COLUMN_VALUE)
-    }, "Expected no active member in describe group results.")
-
-    consumerGroupCommand.close()
+    consumerGroupExecutor.shutdown()
+
+    val (result, succeeded) = TestUtils.waitAndGet(consumerGroupService.describeGroup()) { case (state, assignments) =>
+      val testGroupAssignments = assignments.toSeq.flatMap(_.filter(_.group == group))
+      def assignment = testGroupAssignments.head
+      state == Some("Empty") &&
+        testGroupAssignments.size == 1 &&
+        assignment.consumerId.exists(_.trim == ConsumerGroupCommand.MISSING_COLUMN_VALUE) && // the member should be gone
+        assignment.clientId.exists(_.trim == ConsumerGroupCommand.MISSING_COLUMN_VALUE) &&
+        assignment.host.exists(_.trim == ConsumerGroupCommand.MISSING_COLUMN_VALUE)
+    }
+    val (state, assignments) = result
+    assertTrue(s"Expected no active member in describe group results, state: $state, assignments: $assignments",
+      succeeded)
```

Comment:
This makes the code slightly more complicated, but it gives more information on failure. So I think it is better this way.